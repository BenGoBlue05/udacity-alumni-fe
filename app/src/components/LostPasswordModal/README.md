## LostPasswordModal Component
A component that ...

### Example

```js
<LostPasswordModal />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
