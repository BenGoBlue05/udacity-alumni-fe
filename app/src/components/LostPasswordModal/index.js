import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Layer from 'grommet-udacity/components/Layer';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import Form from 'grommet-udacity/components/Form';
import FormField from 'grommet-udacity/components/FormField';
import Button from 'grommet-udacity/components/Button';
import { LoadingIndicator, ErrorAlert } from 'components';

const LostPasswordModal = ({
  isVisible,
  onClose,
  emailInput,
  onSubmit,
  onChangeEmailInput,
  didSubmit,
  isLoading,
  error,
  onClearError,
}) => (
  <Layer
    closer
    align="center"
    onClose={onClose}
    hidden={!isVisible}
    className={styles.lostPasswordModal}
  >
    <Box pad="medium" className={styles.outerBox}>
      <Box pad="medium" className={styles.innerBox}>
        <Heading align="center">
          Reset Your Password
        </Heading>
        <Heading align="center" tag="h5">
          Please provide the email address that you used when you signed up for the site.
        </Heading>
        <Heading align="center" tag="h5">
          We will send you an email with a link you can use to reset your password.
        </Heading>
      </Box>
      {didSubmit ?
        <Box align="center">
          <Heading align="center" tag="h3">
            Success!  Check your email for instructions to complete the process.
          </Heading>
          <Button
            onClick={onClose}
            label="Close"
            className={styles.button}
          />
        </Box>
      :
        <Form className={styles.form}>
          <Box direction="row" style={{ width: '100%' }}>
            <FormField className={styles.formField}>
              <input
                value={emailInput}
                onChange={onChangeEmailInput}
                placeholder="david@udacity.com"
                type="text"
                name="email"
              />
            </FormField>
            <Button
              primary
              className={styles.button}
              onClick={onSubmit}
              label="Submit"
            />
          </Box>
        </Form>
      }
    </Box>
    {isLoading && <LoadingIndicator message="Submitting" isLoading={isLoading} />}
    {error && <ErrorAlert errors={[error]} onClose={onClearError} />}
  </Layer>
);

LostPasswordModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  emailInput: PropTypes.string,
  onChangeEmailInput: PropTypes.func.isRequired,
  didSubmit: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  error: PropTypes.object,
  onClearError: PropTypes.func.isRequired,
};

export default cssModules(LostPasswordModal, styles);
