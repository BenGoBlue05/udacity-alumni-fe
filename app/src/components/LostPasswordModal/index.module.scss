.outerBox {
  padding: 60px 0 !important;
}

.innerBox {
  padding: 40px 40px !important;
}

.form {
  margin: 40px;
}

.formField {
  @media screen and (max-width: 720px) {
    max-width: 70vw;
  }
}

.button {
  margin-left: 20px;
  @media screen and (max-width: 720px) {
    max-width: 70vw;
    margin-left: 0;
    margin-top: 20px;
  }
}
