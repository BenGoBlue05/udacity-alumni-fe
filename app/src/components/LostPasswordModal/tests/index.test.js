import LostPasswordModal from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<LostPasswordModal />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <LostPasswordModal />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
