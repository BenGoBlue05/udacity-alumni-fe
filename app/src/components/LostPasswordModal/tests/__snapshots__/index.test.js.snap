exports[`<LostPasswordModal /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={true}>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="medium"
    responsive={true}
    tag="div">
    <Box
      announce={false}
      direction="column"
      focusable={true}
      pad="medium"
      responsive={true}
      tag="div">
      <Heading
        align="center"
        tag="h1">
        Reset Your Password
      </Heading>
      <Heading
        align="center"
        tag="h5">
        Please provide the email address that you used when you signed up for the site.
      </Heading>
      <Heading
        align="center"
        tag="h5">
        We will send you an email with a link you can use to reset your password.
      </Heading>
    </Box>
    <Form
      compact={false}
      fill={false}
      pad="none">
      <Box
        announce={false}
        direction="row"
        focusable={true}
        pad="none"
        responsive={true}
        style={
          Object {
            "width": "100%"
          }
        }
        tag="div">
        <FormField
          size="medium"
          strong={false}>
          <input
            name="email"
            placeholder="david@udacity.com"
            type="text" />
        </FormField>
        <Button
          label="Submit"
          primary={true}
          type="button" />
      </Box>
    </Form>
  </Box>
</Layer>
`;
